robot.py

```diff
@@ -53,35 +59,51 @@ def robotInit(self):
 
         # @TODO: Find actual non-placeholder values for the channel IDs
         self.wings = Wings(
-            wpilib.DoubleSolenoid(0, 1),
             wpilib.DoubleSolenoid(2, 3),
+            wpilib.DoubleSolenoid(4, 5),
         )
 
+        self.brake = wpilib.DoubleSolenoid(0, 1)
+
         self.driver = wpilib.XboxController(0)
         self.operator = wpilib.XboxController(1)
 
         self.auto_exec = iter([])
 
         self.gyro = wpilib.ADXRS450_Gyro()
 
-        self.vision_socket = network.VisionSocket()
+        if Robot.isReal():
+            self.vision_socket = network.VisionSocket()
+            self.switch_configuration = wpilib.DriverStation.getInstance().getGameSpecificMessage()
+        else:
+            self.vision_socket = network.MockSocket()
+            self.switch_configuration = "Sim Mode"
+
         self.vision_socket.start()
         self.timer = 0
 
     def robotPeriodic(self):
         if self.timer % 100 == 0:
             print(self.vision_socket.get_angle(1.0))
+            print("ID: {}".format(self.vision_socket.get_id()))
             print("is bound: {}".format(self.vision_socket.is_bound()))
         self.timer += 1
 
     def teleopInit(self):
         print("Teleop Init Begin!")
+        self.switch_configuration = wpilib.DriverStation.getInstance().getGameSpecificMessage()
```

Comment:
You're avoiding this if not `Robot.isReal()` above, should this be conditional / moved into a separate function with that conditional?

Reply:
Yes, it should, although I'm gonna return enums instead of a string in this function

Reply:
Oh, hm... `Robot.isReal()` is false when in sim mode, so we'd end up breaking the simulator if we try to use it.